File: BubbleSort.java
import java.util.Arrays;

class BubbleSort {
    void bubbleSort(int arr[])
    {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                    if (arr[j] > arr[j + 1]) {
                    int tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                }
            }

        }
    }
 
    public static void main(String args[])
    {
        BubbleSort ob = new BubbleSort();
        int arr[] = { 27, 84, 89, 71, 99, 35, 50 };
        ob.bubbleSort(arr);
        System.out.println("Sorted array =>");
        System.out.println(Arrays.toString(arr));
    }
}

File: BubbleSort.js
/** 
 * This function will accept an array and
 * Sort its elements in ascending order and return the sorted array
 * @param {Array} arr array with elements of any data type
 * @returns {Array} array with elements sorted in ascending order
 */
const BubbleSort = (arr) => {
  const n = arr.length;
  let swapped;

  do {
    swapped = false;
    for (let i = 0; i < n - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        // Swap arr[i] and arr[i+1] if they are in the wrong order
        const temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;
        swapped = true;
      }
    }
  } while (swapped);

  return arr;
};

export { BubbleSort };
